backend/routers/procurement.py

```
@@ -0,0 +1,427 @@
+    if not supplier:
+        raise HTTPException(status_code=404, detail="Supplier not found")
+    
+    # Generate PO number
+    po_count = db.query(func.count(models.PurchaseOrder.id)).filter(
+        models.PurchaseOrder.company_id == current_user.company_id
+    ).scalar()
+    po_number = f"PO-{date.today().strftime('%Y%m')}-{po_count + 1:05d}"
+    
+    # Calculate totals
+    subtotal = sum(line.quantity * line.unit_price for line in order.lines)
+    tax_amount = subtotal * Decimal(0.16)  # 16% VAT default
```

Ruff (0.14.4)

`Decimal()` called with float literal argument

Replace with string literal

(RUF032)